plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/nessie/NessieIcebergClient.java

```diff
@@ -0,0 +1,432 @@
+                    format("Cannot rename table '%s' to '%s': ref '%s' no longer exists", from.getTableName(), to.getTableName(), getReference().getName()), e);
+        }
+        catch (BaseNessieClientServerException e) {
+            throw new TrinoException(ICEBERG_CATALOG_ERROR,
+                    format("Cannot rename table '%s' to '%s': ref '%s' is not up to date", from.getTableName(), to.getTableName(), getReference().getName()), e);
+        }
+        catch (HttpClientException ex) {
+            throw new TrinoException(ICEBERG_CATALOG_ERROR, ex);
+        }
+    }
+
+    public void commitTable(TableMetadata metadata, SchemaTableName tableName, String metadataLocation, String user)
+    {
+        getReference().checkMutable();
+        try {
+            ImmutableIcebergTable.Builder newTableBuilder = ImmutableIcebergTable.builder();
+            Snapshot snapshot = metadata.currentSnapshot();
+            long snapshotId = snapshot != null ? snapshot.snapshotId() : -1L;
+            IcebergTable newTable = newTableBuilder
+                    .snapshotId(snapshotId)
+                    .schemaId(metadata.currentSchemaId())
+                    .specId(metadata.defaultSpecId())
+                    .sortOrderId(metadata.defaultSortOrderId())
+                    .metadataLocation(metadataLocation)
+                    .build();
+
+            Branch branch = getNessieApi().commitMultipleOperations()
+                    .operation(Operation.Put.of(NessieIcebergUtil.toKey(tableName), newTable))
+                    .commitMeta(buildCommitMeta(user, format("Trino Iceberg add table %s", tableName)))
+                    .branch(getReference().getAsBranch())
+                    .commit();
+            getReference().updateReference(branch);
+        }
+        catch (NessieNotFoundException e) {
+            throw new TrinoException(ICEBERG_COMMIT_ERROR, format("Cannot commit: ref '%s' no longer exists", getReference().getName()), e);
+        }
+        catch (NessieConflictException e) {
+            throw new TrinoException(ICEBERG_COMMIT_ERROR, format("Cannot commit: ref hash is out of date. Update the ref '%s' and try again", getReference().getName()), e);
+        }
+    }
+
+    static class UpdateableReference
+    {
+        private Reference reference;
+        private final boolean mutable;
+
+        // required for dependency injection
+        public UpdateableReference()
+        {
+            this(Branch.of("main", null), false);
+        }
+
+        /**
+         * Construct a new {@link UpdateableReference} using a Nessie reference object and a flag
+         * whether an explicit hash was used to create the reference object.
+         */
+        public UpdateableReference(Reference reference, boolean hashReference)
+        {
+            this.reference = reference;
+            this.mutable = reference instanceof Branch && !hashReference;
+        }
+
+        public boolean refresh(NessieApiV1 api)
```

Comment:
use `nessieApi` from the outer class context

Reply:
done